### DISCUSSION TITLE
`Payload 3`: Is access control different than in `v2`?

### DISCUSSION THREAD

**Discussion by Fitch24** [payloadcms/payload#9924] [Q&A]
When using Access Control (for collections) in `v2` I can throw errors to return (in api response) specified status codes and error messages:
```typescript
import type { CollectionConfig } from "payload/types";
import { APIError } from "payload/errors";

export const Posts: CollectionConfig = {
  slug: "posts",
  fields: [
    {
      name: "title",
      type: "text",
      required: true,
    },
    {
      name: "content",
      type: "textarea",
      required: true,
    },
    {
      name: "author",
      type: "relationship",
      relationTo: "users",
      hasMany: false,
      required: false,
    },
  ],
  access: {
    create: ({ req }) => {
      if (!req.user) {
        throw new APIError("You need to sign in to create posts", 401);
      }
      if (!(req.user.permissions as string[]).includes("CREATE_POSTS")) {
        throw new APIError("You don't have permissions to create posts", 403);
      }
      return true;
    },
  },
};
// It's just for example. In real project I have more complex logic and custom errors inherited from APIError
```
I tried the same way in `v3` (latest v3.6.0) and api works the same as `v2` (and as I expected). But admin panel was completely broken: any page (from `/admin` route) throws errors during Next.js rendering stage because of calling all access methods from all collections (for determine access privileges, I think).
![admin_panel](https://github.com/user-attachments/assets/edb4b150-885c-4c3e-b041-864604e56fc1)

Is it possible to throw errors in access methods in `v3`? Or I originally used anti-pattern in `v2` and shouldn't throw any errors in collection access methods?

**Comment by matthijs166** [comment]
Indeed, you use an antipattern, the access control functions need to return a boolean.
https://payloadcms.com/docs/access-control/collections#create

I think it is because of the server side rendering in Next.js, the error is thrown server side while rendering and fetching the page.
Don't think that server side rendering was happening in payload 2.0.
